Source/Plugins/Core/com.equella.base/src/com/tle/common/searching/Search.java

```diff
@@ -93,4 +93,17 @@ public SortField getSortField(boolean reverseTheDefault) {
    * @return True if server time zone is used or otherwise false.
    */
   boolean useServerTimeZone();
+
+  /**
+   * The background of creating this method is to support the Lucene query built in New Advanced
+   * search page.
+   *
+   * <p>The result of calling this method is a String in Lucene query syntax or null. e.g. name:OEQ
+   * AND year:[2021-11-01 TO * ] AND (city:Hobart OR city:Sydney) AND message:"hello world"
+   *
+   * @return A string is Lucene query syntax or null.
+   */
+  default String getCustomLuceneQuery() {
+    return null;
+  }
```

Comment:
Could we push with the new direction of Optional?

```suggestion
  default Optional<String> getCustomLuceneQuery() {
    return Optional.empty();
  }
```

If doing this, make sure to update the JavaDoc too.